File: src/utils/functions/mad.js

export function mad () {

    //calculate the mean of inputs
    var mean=0
    for (var i = 0; i < arguments.length; i++) {
    	mean += arguments[i]
    }
  	mean = mean/arguments.length
  
    //sum up the difference between each number and mean
    var deviation=0
    var sum=0
    for (var i = 0; i < arguments.length; i++) {
        deviation=arguments[i]-mean
        if (deviation<0){
            deviation = -deviation
        }
        sum+=deviation
    }

    //divide the sum of difference by number of inputs
    return sum/arguments.length
  }

File: app/functions/mad.py
def mad(*arg):
  # calculate the mean of inputs
  mean = 0
  for i in range(0,len(arg)):
    mean += arg[i]
  mean = mean / len(arg)

  # sum up the difference between each number and mean
  deviation = 0
  sum = 0
  for i in range(0,len(arg)):
    deviation = arg[i] - mean
    if (deviation < 0):
      deviation = -deviation
    sum += deviation
    
  # divide the sum of difference by number of inputs
  return sum / len(arg)